internal/shareddefaults/shared_config_resolve_home_go1.12.go

```diff
@@ -0,0 +1,13 @@
+//go:build go1.12
+// +build go1.12
+
+package shareddefaults
+
+import (
+	"os"
+)
+
+func userHomeDir() string {
+	home, _ := os.UserHomeDir()
```

Comment:
This does not fix the problem with #4444, since `os.UserHomeDir` is also based on `$HOME`.
Test program to demonstrate:
```go
// go build test.go
// env -i ./test
package main                                                                    
                                                                                
import (                                                                        
        "fmt"                                                                   
        "os"                                                                    
)                                                                               
                                                                                
func main() {                                                                   
        home, err := os.UserHomeDir()                                           
        fmt.Println(home,  err)                                            // prints "$HOME not defined" if variable is empty
}  
```

Better solution:
```go
package main                                                                    
                                                                                
import (                                                                        
        "fmt"                                                                   
        "os/user"                                                               
)                                                                               
                                                                                
func main() {                                                                   
        user, err := user.Current()                                             
        fmt.Println(user.HomeDir, err)                  // Prints home directory even when running under env -i or $HOME unset.
}                                                                               
```
The `user.Current()` is based on `getpwent` and thus is independent of the `$HOME` variable.